apps/studio/data/analytics/warehouse-access-tokens-create-mutation.ts

```diff
@@ -0,0 +1,45 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query'
+import { analyticsKeys } from './keys'
+import { post } from 'data/fetchers'
+
+type CreateWarehouseAccessToken = {
+  description: string
+}
+async function createWarehouseAccessToken(ref: string, payload: CreateWarehouseAccessToken) {
+  const res = await post(`/platform/projects/{ref}/analytics/warehouse/access-tokens`, {
+    params: {
+      path: { ref },
+    },
+    body: payload,
+  } as any)
+  // TODO: remove type casts when openapi client generates correct types
+
+  return res as any as {
+    id: string
+    token: string
+    scopes: string
+    inserted_at: string
+    description?: string
+  }
+}
+
+type CreateAccessTokenArgs = {
+  projectRef: string
+  onSuccess: (data: Awaited<ReturnType<typeof createWarehouseAccessToken>>) => void
+}
+
+export function useCreateWarehouseAccessToken({ projectRef, onSuccess }: CreateAccessTokenArgs) {
```

Comment:
i just realised this - could we follow the way we write our react query hooks as per what's in our template here?
https://github.com/supabase/supabase/blob/master/apps/studio/data/__templates/resource-update-mutation.ts#L27

or can refer to the other RQ hooks - the templates here ensure proper error handling as well cause i noticed that the S3 credentials one would throw an error if the deletion request somehow failed. also for consistency as well 🙏  will apply for mutations and queries

Reply:
no worries about the S3 credentials one though - i'm refactoring them in another PR

Reply:
@joshenlim perfect situation that an eslint plugin would be great 😉